docs/architecture.rst

```diff
@@ -0,0 +1,154 @@
+How ppb Fits Together
+===========================================================
+
+This document is an overall reference of what ppb is, how it's organized, and
+how to use the various pieces.
+
+The Conceptual Necessities
+-----------------------------------------------------------
+.. _arch-the-basics:
+
+1. A basic outline of the core elements of using ppb:
+    1. The GameEngine
+    2. Scenes
+    3. Sprites
+    4. Events, EventTypes, Event Handlers
+    5. Flags?
+    6. Assets?
+
+This section should be light and to the point: It's basically a technical
+reference version of getting started and link to more detailed descriptions
+further in this document.
+
+Things not included in this list but will be important to document:
+
+1. Systems and Systemslib
+2. GameObject
+3. Time?
+4. Camera
+
+Events
+-----------------------------------------------------------
+.. _arch-events:
+
+1. What is an event
+2. Event classes
+    1. dataclasses/attr classes
```

Comment:
reminder that using `dataclasses` isn't a requirement. it's just an easy way to make a data bag.

Reply:
That's as much a suggestion/how we do it as anything. But will keep in mind.